### ISSUE TITLE
early_stopping() min_delta kwarg unexpected

### ISSUE THREAD

**Issue by kylebeni** [microsoft/LightGBM#5697]
## Description
In lightgbm 3.3.5, passing `min_delta` with a float value should work, but is throwing an "unexpected keyword argument" error. Removing the kwarg works, but the default value is 0.0 which means early stopping won't occur in that case.

## Reproducible example
```python
import lightgbm

print(lightgbm.__version__)  # confirm version is 3.3.5

lightgbm.early_stopping(
                    stopping_rounds=100,
                    first_metric_only=False,
                    verbose=True,
                    min_delta=0.00001,
                )
```

Returns:
```python
3.3.5
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Untitled-1.ipynb Cell 1 in <cell line: 5>()
      [1](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=0) import lightgbm
      [3](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=2) print(lightgbm.__version__)
----> [5](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=4) lightgbm.early_stopping(
      [6](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=5)                     stopping_rounds=100,
      [7](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=6)                     first_metric_only=False,
      [8](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=7)                     verbose=True,
      [9](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=8)                     min_delta=0.00001,
     [10](vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W0sdW50aXRsZWQ%3D?line=9)                 )

TypeError: early_stopping() got an unexpected keyword argument 'min_delta'
```

## Environment info
```python
import platform as pf
pf.machine(), pf.version(), pf.platform(), pf.uname(), pf.system(), pf.processor()

('x86_64',
 '#1 SMP Debian 5.10.162-1 (2023-01-21)',
 'Linux-5.10.0-21-cloud-amd64-x86_64-with-glibc2.29',
 uname_result(system='Linux', node='kmb', release='5.10.0-21-cloud-amd64', version='#1 SMP Debian 5.10.162-1 (2023-01-21)', machine='x86_64', processor=''),
 'Linux',
 '')
```

LightGBM version or commit hash: 3.3.5

Command(s) you used to install LightGBM

```shell
pip install lightgbm
```

virtualenv on Linux cloudvm, tested in both python scripts and Jupyter notebooks with lightgbm 3.3.5.

## Additional Comments

None.

**Comment by jmoralez** [comment]
Hi @kylebeni, thanks for using LightGBM. You're probably looking at the `latest` version of the documentation. If you look at [v3.3.5](https://lightgbm.readthedocs.io/en/v3.3.5/pythonapi/lightgbm.early_stopping.html#lightgbm.early_stopping) you can see that `min_delta` isn't present in that release, it will be available in the next major release (v4.0.0) which you can track in #5153.

**Comment by kylebeni** [comment]
Thanks for the quick response @jmoralez. In the future the delineation between dev (latest) and prod docs could be clearer. 

It isn't obvious that "latest" means "unreleased". Thanks.
